### ISSUE TITLE
advertise client urls accepts empty hosts

### ISSUE THREAD

**Issue by heyitsanthony** [coreos/etcd#8379]
A command line misconfiguration from irc:
```sh
$ SPELLEDOK="localhost"
$ etcd --listen-client-urls=http://127.0.0.1:2379 --advertise-client-urls=http://$SPELEDOK:2379 &
$ etcdctl member list
8e9e05c52164694d: name=default peerURLs=http://localhost:2380 clientURLs=http://:2379 isLeader=true
```

etcd should probably detect empty advertise hostnames and exit with an error.

/cc @kbrwn

**Comment by shubheksha** [comment]
Hi, I'd like to work on this issue if it's not taken already. This would be my first contribution so I'd appreciate some help regarding where to get started. 😄

**Comment by gyuho** [comment]
@shubheksha That would be great. Please start from https://github.com/coreos/etcd/pull/8384 as reference. And let us know if you need more help.

Thanks!

**Comment by xiang90** [comment]
fixed by #8786
